fix(download/mirrors) update instructions for mirror maintainers (#8165)
jenkins-infra/jenkins.io: content/download/mirrors.adoc

- infrastructure
:author: rtyler
---

> Jenkins packages and plugins are served by the web service https://get.jenkins.io (or its legacy alias mirrors.jenkins.io).

Reviewer: ```suggestion
Jenkins packages and plugins are served by the web service link:https://get.jenkins.io[get.jenkins.io] (or its legacy alias mirrors.jenkins.io).
```

> The web service https://get.jenkins.io relies on a network of mirror servers provided by generous sponsors of the Jenkins project all over the world.

Reviewer: ```suggestion
The web service link:https://get.jenkins.io[get.jenkins.io] relies on a network of mirror servers provided by generous sponsors of the Jenkins project all over the world.
```

> When a user download a plugin or a file, the web service https://get.jenkins.io redirects (transparently) the user to the closest mirror.

Reviewer: ```suggestion
When a user download a plugin or a file, the web service link:https://get.jenkins.io[get.jenkins.io] redirects (transparently) the user to the closest mirror.
```

> The web service https://get.jenkins.io is based on the Open Source mirror redirector https://github.com/etix/mirrorbits.

Reviewer: ```suggestion
The web service link:https://get.jenkins.io[get.jenkins.io] is based on the Open Source mirror redirector link:https://github.com/etix/mirrorbits[mirrorbits].
```

The list of mirrors (and their locations) is available on the following page: link:https://get.jenkins.io/index.html?mirrorstats[mirror status page].

> ## How to Help the Jenkins Project by Running a Mirror?

Reviewer: Overall feedback:

There's a lot of text here it would be good to either provide an issue form template or an example input so someone can copy paste and edit if required.

an issue form would be best

Author: I thought the same when opening the PR: this page has too many textual information and it might not be the best place for the "reference documentation for mirror maintainers".

WDYT if we move the "list" in a markdown in jenkins-infra/documentation as a dedicated page and link it there instead? 

I don't mind the issue form template, but it would be only cover the case of "adding a new mirror".

Author: Oh I see, the issue is not having "too much data" but more the rendering (e.g. hard to browse or read at first sight). Ack, then let me try something then. thanks!

Author: @timja done, I've implemented the list as subsections. How do it fare?

Download mirror servers are always welcome to reduce the load on individual servers, to spread the bandwidth costs among multiple servers and to provide locations close to users.

If you provide (or want to provide) a download mirror server, you'll have to fulfill the requirements below.

> We provide the link:https://github.com/jenkins-infra/helpdesk[GitHub "jenkins-infra" helpdesk] were you can open issues to add/change/delete a mirror or if you have any problem.

Reviewer: ```suggestion
We provide the link:https://github.com/jenkins-infra/helpdesk[GitHub "jenkins-infra" helpdesk] where you can open issues to add/change/delete a mirror or if you have any problem.
```

Mirror requirements:

* An HTTP server to serve files for the end users with:

> ** A public IPv4 and IPv6 (we recommend both). You can have multiple IPs (see below).

Reviewer: ```suggestion
** A public IPv4 and IPv6 address (we recommend both). You can have multiple IPs (see below).
```

** A valid and resolvable public domain name pointing to your public IPs (we recommend `A` and `AAAA` DNS records). Multi-valued DNS records are valid if you have many IPs.
** Enforced HTTPS with a valid TLS certificate

> * A storage of at least 300Gb to to store ~1 year of files served by the HTTP server:

Reviewer 2: ```suggestion
* A storage of at least 300Gb to store ~1 year of files served by the HTTP server:
```

> ** As a data point, the "reference" download server, which is https://archives.jenkins.io, uses around ~600Gb with **all** of the files.

Reviewer: ```suggestion
** As a data point, the "reference" download server, which is link:https://archives.jenkins.io[archives.jenkins.io], uses around ~600Gb with **all** of the files.
```

> ** 300Gb of data is enough to provide artifacts published for 1 year (recommended time window). You may clean artifacts older than 1 year (provided by the archives.jenkins.io).

Reviewer: ```suggestion
** 300Gb of data is enough to provide artifacts published for 1 year (recommended time window). You may clean artifacts older than 1 year (provided by link:https://archives.jenkins.io[archives.jenkins.io]).
```

> * A rsync or FTP server to allow https://get.jenkins.io to scan your mirror regularly:

Reviewer: ```suggestion
* A rsync or FTP server to allow link:https://get.jenkins.io[get.jenkins.io] to scan your mirror regularly:
```

** Should serve the same data as the HTTP server

> ** You can restrict inbound requests using IPs: we provide an API endpoint at https://reports.jenkins.io/infrastructure/v2/index.json with our rsync/ftp outbound scanning IPs.

Reviewer: ```suggestion
** You can restrict inbound requests using an IP allowlist, we provide an API endpoint at link:https://reports.jenkins.io/infrastructure/v2/index.json[reports.jenkins.io/infrastructure/v2/index.json] with our rsync/ftp outbound IPs.
```

*** Check the JSON list at `"get.jenkins.io".outbound_ips[]`
*** Our scanning IPs do not change often (once a year usually) but we update the JSON daily in case of any change

* A rsync routine to keep your data up to date:

> ** The data source is rsync://archives.jenkins.io without authentication but we need your outbound IPs to restrict access.

Reviewer: ```suggestion
** The data source is `rsync://archives.jenkins.io` without authentication but we need your outbound IPs to restrict access.
```

** We recommend a (differential) data update at least hourly
** The file can be deleted if they are more than 1 year old to avoid filling your disks, unless you have at least 600 Gb of data available

* A contact email if we need to reach out to the mirror's administrators for advertising configuration change or errors reported by users.
** If you wish for us to keep this email privately, please send it to jenkins-infra-team@googlegroups.com

* The outbound IP(s) used by your rsync server to update data (so we can apply restrictions)

> ** We store these IPs in https://github.com/jenkins-infra/infra-reports/blob/main/jenkins-infra-data/get-jenkins-io-data.json

Reviewer: ```suggestion
** We store these IPs in link:https://github.com/jenkins-infra/infra-reports/blob/main/jenkins-infra-data/get-jenkins-io-data.json[get-jenkins-io-data.json]
```
